### PR TITLE
Add setGetSerialNumber

### PR DESCRIPTION
**Pull request by joel16**
Title. Let me know if I need to make any changes.

### CODE REVIEW

File: nx/include/switch/services/set.h [hunk 1]
```diff
@@ -66,3 +66,6 @@ void setsysExit(void);
 
 /// Gets the current system theme.
 Result setsysGetColorSetId(ColorSetId* out);
+
+/// Gets the serial number.
```
Comment (fincs) [inline nx/include/switch/services/set.h]: Should document the parameters and mention that the output buffer needs to be able to hold at least 0x18 bytes.

**Comment by yellows8** [comment]
setsys{name}

File: nx/include/switch/services/set.h [hunk 2]
```diff
@@ -66,3 +66,9 @@ void setsysExit(void);
 
 /// Gets the current system theme.
 Result setsysGetColorSetId(ColorSetId* out);
+
+/**
+ * @brief Gets the system's serial number.
+ * @param serial Pointer to output the serial to. (The buffer size needs to be at least 0x18 bytes)
```
Comment (yellows8) [inline nx/include/switch/services/set.h]: *0x19

File: nx/source/services/set.c [hunk 3]
```diff
@@ -346,3 +346,38 @@ Result setsysGetColorSetId(ColorSetId* out)
     return rc;
 
 }
+
+Result setsysGetSerialNumber(char *serial) {
+    IpcCommand c;
+    ipcInitialize(&c);
+
```
Comment (yellows8) [inline nx/source/services/set.c]: Add: if (serial) memset(serial, 0, 0x19);

Comment (joel16) [reply nx/source/services/set.c]: Should I change the declaration as well as the memcpy to 0x19 too?

Comment (yellows8) [reply nx/source/services/set.c]: This is to make sure the string is NUL-terminated.

Comment (joel16) [reply nx/source/services/set.c]: Yes I was aware of that, but I guess the result buffer should be fine then.

**Pull request merged**